### PR TITLE
Add gunmod recipes.

### PR DESCRIPTION
**Pull request by Rail-Runner**
Add recipes for following items:
1. Adjustable stock
2. Ported barrel
3. Bipod
4. Rapid blowback
5. Extended magazine
6. Double magazine
7. Forward grip
8. Improved iron sights
9. Underbarrel laser sight
10. Muzzle brake
11. Ergonomic grip
12. Pistol stock
13. Rail laser sight
14. Recoil stock
15. Spare magazine
16. Firearm waterproofing
17. Bow sight
18. Bow stabilizer
19. Arrow rest
20. Rail-mounted crossbow

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
I'm opposed to stuff like scopes/fancy sights being craftable without a proper machine shop; the tolerances required are far too precise for survivors to make with what we have available right now.

The rest of it is cool, though some of them (like the magazines and waterproofing) should probably require a gunsmithing kit to make.

**Comment by Darkyhard** [comment]
Actually you can build a simple homemade telescope if you have some lenses. Like this: http://blog.imaginechildhood.com/imagine-childhood/2009/07/telescoping.html

However, I'm not exactly sure if it is possible to make a scope like this.

**Comment by PropaneSoup** [comment]
I think creating a "homemade" rifle scope is reasonable, obviously with some lower stats.

**Comment by Rivet-the-Zombie** [comment]
The issue isn't making a tube that magnifies images; it's making a perfectly calibrated tube that magnifies images, has proper markings to allow you to actually aim with it, can be affixed to a weapon and not get in the way of its operation, can be adjusted once it's solidly mounted to a weapon so as to be zeroed in on its point of aim, and this all has to be something built solidly enough that firing the weapon won't damage it or knock it back out of alignment.

No, it's not reasonable.

**Comment by kevingranade** [comment]
I strongly suspect that a homemade telescope on a rifle is going to shake
itself to pieces in short order, and be worthless for aiming even sooner.
There are probably some intermediate sights that have fallen out of favor
since scopes are so dominant now that it would be cool to dredge up, but
that's a whole new item rather than a new recipe.

File: data/json/recipes/recipe_weapon.json [hunk 1]
```diff
@@ -2221,6 +2221,873 @@
...
+  "time": 30000,
+  "reversible": false,
+  "autolearn": false,
+  "book_learn": [[ "textbook_mechanics", 3 ] , [ "manual_pistol", 4] , [ "manual_shotgun", 4] , [ "manual_smg", 4] , [ "manual_rifle", 4]],
+  "tools": [
+    [
+      [ "large_repairkit", 75 ]
+    ],
+    [
+      [ "swage", -1 ]
```
Comment (KA101) [inline data/json/recipes/recipe_weapon.json]: OK, ported barrel here.  IIRC a swage & die set can put some amount of rifling into the pipe (whether it's at all realistic to use that rifled pipe as a barrel is a separate question), but isn't going to actually bore out the ports.  Might be worth adding a gunsmith kit here.

File: data/json/recipes/recipe_weapon.json [hunk 2]
```diff
@@ -2221,6 +2221,873 @@
...
+      [ "pipe", 1 ],
+      [ "steel_chunk", 2 ]
+    ],
+    [
+      [ "plastic_chunk", 3 ]
+    ]
+  ]
+},{
+  "type" : "recipe",
+  "result": "blowback",
```
Comment (KA101) [inline data/json/recipes/recipe_weapon.json]: Hmm.  Thinking rapid-blowback mechanisms might be machine-shop territory.

File: data/json/recipes/recipe_weapon.json [hunk 3]
```diff
@@ -2221,6 +2221,873 @@
...
+    [
+      [ "large_repairkit", 15 ]
+    ]
+  ],
+  "components": [
+    [
+      [ "scrap", 1 ]
+    ],
+    [
+      [ "alloy_sheet", 1 ]
```
Comment (KA101) [inline data/json/recipes/recipe_weapon.json]: Uh, is there *that* much superalloy that folks can use it to make lighter-weight match triggers?

(If so, I'd think we have a surplus of superalloy.  If building a trigger assembly is reasonable with field tools, that's impressive, as I'd think "shop", but then I'm not a machinist.)

File: data/json/recipes/recipe_weapon.json [hunk 4]
```diff
@@ -2221,6 +2221,873 @@
...
+      [ "spring", 1 ]
+    ]
+  ]
+},{
+  "type" : "recipe",
+  "result": "stabilizer",
+  "category": "CC_WEAPON",
+  "subcategory": "CSC_WEAPON_MODS",
+  "skill_used": "mechanics",
+  "skills_required": [[ "electronics", 5 ]],
```
Comment (KA101) [inline data/json/recipes/recipe_weapon.json]: I'm perfectly content to just block the gyro-stabilizer altogether (shop/clean-room territory) but at minimum it ought to require a good working knowledge of guns to understand how it'd counter the recoil.  Figure adding gun 5-6 or so in the required-skills list.

**Comment by KA101** [comment]
OK, yeah, if scopes aren't practical to craft--and they aren't--I'm hard pressed to justify crafting laser or holo/red-dot sights.  Those seem like they'd hit the same problems.

(And perhaps laser weapons shouldn't be craftable from scratch, too.)

File: data/json/recipes/recipe_weapon.json (hunk 3 above)
Comment (Rivet-the-Zombie) [inline data/json/recipes/recipe_weapon.json]: Yeah this is definitely a 'shop' thing. There's no way you're making a match-grade trigger assembly (or any other high quality internal mechanisms) without one.

File: data/json/recipes/recipe_weapon.json (hunk 4 above)
Comment (Rivet-the-Zombie) [inline data/json/recipes/recipe_weapon.json]: I agree. I'm pretty sure that whatever technology is inside the gyroscope stabilizer is the sort of thing that you're not going to replicate without some seriously fancy manufacturing equipment and a lot of very specialized tools.

File: data/json/recipes/recipe_weapon.json (hunk 2 above)
Comment (Rivet-the-Zombie) [inline data/json/recipes/recipe_weapon.json]: This is about as complex a gunmod as I could see somebody crafting without a machine shop; there's a lot of firearms out there that can be modified to fire full-auto without major rebuilding, and I'm pretty sure that's what this gunmod is supposed to represent.

**Pull request merged**